File: src/test/java/parser/EquipementsParserTest.java
package parser;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import domain.Equipment;
import domain.FromTo;
import domain.Statistic;
import org.junit.Assert;
import org.junit.Test;

import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.io.Writer;
import java.util.ArrayList;
import java.util.List;

public class EquipementsParserTest {

    private EquipementsParser equipementsParser = new EquipementsParser();
    private String testDir = getClass().getResource("equipments_test.json").getPath();

    @Test
    public void deserializedEquipments() throws FileNotFoundException {
        // When
        List<Equipment> eList = equipementsParser.deserializedEquipments(testDir);

        // Then
        Assert.assertEquals(eList.get(0).get_id(), 14094);
        Assert.assertEquals(eList.get(0).getName(), "Amulette du Strigide");
        Assert.assertEquals(eList.get(0).getLevel(), "200");
        Assert.assertEquals(eList.get(0).getType(), "Amulette");
        Assert.assertEquals(eList.get(0).getImgUrl(), "https://s.ankama.com/www/static.ankama.com/dofus/www/game/items/200/1235.png");
        Assert.assertEquals(eList.get(0).getStatistics().get(0).getVitalite().getMin(), "351");
        Assert.assertEquals(eList.get(0).getStatistics().get(0).getVitalite().getMax(), "400");
    }

    @Test
    public void writeInFileTest() throws IOException {
        // Given
        List<Equipment> equipments = new ArrayList();
        Equipment equipment = new Equipment();
        equipment.set_id(14094);
        equipment.setName("Amulette du Strigide");
        equipment.setLevel("200");
        equipment.setType("Amulette");
        equipment.setImgUrl("https://s.ankama.com/www/static.ankama.com/dofus/www/game/items/200/1235.png");

        List<Statistic> statistics = new ArrayList();
        Statistic statistic = new Statistic();
        statistic.setVitalite(new FromTo("0", "10"));
        statistics.add(statistic);

        equipment.setStatistics(statistics);
        equipments.add(equipment);

        String path = getClass().getResource("data/amulettes_test.json").getPath();
        Writer amulettesWriter = new FileWriter(path);
        Gson gson = new GsonBuilder().setPrettyPrinting().create();

        // When
        equipementsParser.writeInFile(equipments, amulettesWriter, gson);
    }
}

File: src/test/java/parser/EquipementsParserTest.kt
package parser

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import domain.Equipement
import domain.FromTo
import domain.Statistic
import org.junit.Assert
import org.junit.Assert.*
import org.junit.Test
import java.io.File

class EquipementsParserTest {

    @Test
    fun testWriteAmulettes() {
        val equipementsPath = javaClass.getResource("/equipements_test.json").path
        val expected = Equipement(
            _id= 14094,
            name = "Amulette du Strigide",
            level = 200,
            imgUrl = "https://s.ankama.com/www/static.ankama.com/dofus/www/game/items/200/1235.png",
            type = "Amulette",
            statistic = listOf(
                Statistic("vitalite", FromTo(351, 400)),
                Statistic("sagesse", FromTo(31, 40)),
                Statistic("critique", FromTo(4, 6)),
                Statistic("puissance", FromTo(51, 70)),
                Statistic("pa", FromTo(1, 1)),
                Statistic("pourcentResistanceTerre", FromTo(6, 8)),
                Statistic("pourcentResistanceAir", FromTo(6, 8)),
                Statistic("esquivePa", FromTo(4, 6)),
                Statistic("tacle", FromTo(11, 15)),
                Statistic("dommagesCritiques", FromTo(16, 25)),
                Statistic("resistanceCritiques", FromTo(-16, -20))
            ),
            setId = 274
        )

        EquipementsParser.writeAmulettes(equipementsPath)
        val actual = jacksonObjectMapper().readValue<List<Equipement>>(File(javaClass.getResource("/equipements/amulettes.json").path)).first()

        assertEquals(expected, actual)
    }
}